docs/source/optim.rst

```diff
@@ -279,3 +279,30 @@ and start to collect SWA averages of the parameters at epoch 160:
 >>> torch.optim.swa_utils.update_bn(loader, swa_model)
 >>> # Use swa_model to make predictions on test data 
 >>> preds = swa_model(test_input)
+
+Functional Optimizer API
+---------------------------
+
+If you would like to update your parameters in a pure functional way, you can
+consider using the pure functional optimizer API. The functional APIs for
+optimizer allow you to only do the computation for the parameters you passed in,
+and inplace update the parameters accordingly.
+
+Note that in order to use the functional API, you should maintain the states
+for different algorithms on your own, and update some part of your non-tensor
+states before or after the call to this API (e.g. To use
+:func:`torch.optim.functional.adagrad`, `state_steps` for each parameter need
+to be updated before call to :func:`torch.optim.functional.adagrad`)
```

Comment:
nit: simplifying a little
```suggestion
Note that you need to maintain the states of a functional optimizer by yourself.
In particular, you need to update some part of your non-tensor states before or
after the call to this API (for instance, `state_steps` for each parameter need to
be updated before call to :func:`torch.optim.functional.adagrad`)
```

Reply:
by the way, I do not see the instructions on how to update the states in `torch.optim.functional.adagrad` -- am i missing them?

Reply:
I added `tensor states` together with parameters in the first paragraph to indicate they are updated inplace inside the functional API. Do you think this is enough for instructions?